### DISCUSSION TITLE
Appending conditional CSS classes?

### DISCUSSION THREAD

**Discussion by JohnFoxtrot** [vercel/next.js#39575] [Help]
Came across this when creating an animated dropdown for a navbar.

In a strict React implementation, an inline if/else statement can be used with an onClick toggle to set and remove CSS animation styles. In order to provide a default styling (with no animation) for when state is null, a class can be added before the inline if/else operation:

```
<div className={`navbar ${reactState ? "expanded" : "collapsed"}`}>
```

How do I replicate this in NextJS?

I can't find anything in the documentation and have blindly attempted the following (unsuccessfully):

```
<div className={styles.navbar (reactState ? `${styles.expanded}` : `${styles.collapsed}`)}>
```
```
<div className={styles.navbar [reactState ? `${styles.expanded}` : `${styles.collapsed}`]}>
```
```
<div className={styles.navbar `${reactState ? `${styles.expanded}` : `${styles.collapsed}`}`}>
```

The only success I've had is with the following, which seems like overkill:

```
<div className={
  reactState != null ?
    (reactState ? `${styles.navbar} ${styles.expanded}` 
    : `${styles.navbar} ${styles.collapsed}`)
  : styles.navbar
}>
```

I'm clearly not fully understanding how NextJS handles React styling. Any help would be appreciated.

**Comment by SukkaW** [comment]
Have you tried this library to compose className?
https://npm.im/clsx

**Comment by gaston-flores** [comment]
The `styles.*` generated by CSS modules are simply class names. You can interpolate them in a way similar to your first example:

```jsx
<div className={`${styles.navbar} ${reactState ? styles.expanded : styles.collapsed}`}>
```

[Here's it in action](https://codesandbox.io/s/dazzling-hofstadter-cejnzc?file=/components/Navbar.component.js). 

The library @SukkaW suggests makes this a bit nicer, you can use it outside of next as well to compose your classes:

```jsx
<div className={clsx(styles.navbar, reactState ? styles.expanded : styles.collapsed)}>
```
